layerfetcher/layers_test.go

```diff
@@ -198,24 +199,114 @@ func Test_receiveLayerHash(t *testing.T) {
 	hashRes := RandomHash()
 	// test happy flow - get 4 responses
 	l.receiveLayerHash(context.TODO(), 1, net.peers[0], numOfPeers, hashRes.Bytes(), nil)
-	assert.Equal(t, net.sendCalled, 0)
+	assert.Equal(t, 0, net.sendCalled)
 
 	// test aggregation by hash
 	hashRes2 := RandomHash()
 	for i := 1; i < numOfPeers; i++ {
 		l.receiveLayerHash(context.TODO(), 1, net.peers[i], numOfPeers, hashRes2.Bytes(), nil)
 	}
+	assert.Equal(t, 2, net.sendCalled)
 
-	assert.Equal(t, net.sendCalled, 2)
-
-	//test error flow
+	// test error flow
 	l.receiveLayerHash(context.TODO(), 1, net.peers[0], numOfPeers, hashRes.Bytes(), nil)
 
 	for i := 1; i < numOfPeers; i++ {
 		l.receiveLayerHash(context.TODO(), 1, net.peers[i], numOfPeers, nil, fmt.Errorf("error"))
 	}
 	// no additional sends should happen
-	assert.Equal(t, net.sendCalled, 2)
+	assert.Equal(t, 2, net.sendCalled)
+
+	// test partial empty layer response
+	l.receiveLayerHash(context.TODO(), 1, net.peers[0], numOfPeers, hashRes.Bytes(), nil)
+	l.receiveLayerHash(context.TODO(), 1, net.peers[1], numOfPeers, emptyHash.Bytes(), nil)
+	for i := 2; i < numOfPeers; i++ {
+		l.receiveLayerHash(context.TODO(), 1, net.peers[i], numOfPeers, hashRes2.Bytes(), nil)
+	}
+	// not considered zero-block layer because there are 2 known hashes
+	assert.Equal(t, 4, net.sendCalled)
+
+	// test empty layer response
+	l.receiveLayerHash(context.TODO(), 1, net.peers[0], numOfPeers, nil, fmt.Errorf("error"))
+	for i := 1; i < numOfPeers; i++ {
+		l.receiveLayerHash(context.TODO(), 1, net.peers[i], numOfPeers, emptyHash.Bytes(), nil)
+	}
+	// zero-block layer should not incur additional send
+	assert.Equal(t, 4, net.sendCalled)
+
+	// test two many errors
```

Comment:
"two" or "many"?

Reply:
too many :D

Reply:
it tests the case where numErrors > peers/2, then it errors out without doing anything.